src/internet/radiobrowser/radiobrowserservice.h

```diff
@@ -0,0 +1,145 @@
+
+   You should have received a copy of the GNU General Public License
+   along with Clementine.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef INTERNET_RADIOBROWSER_RADIOBROWSERSERVICE_H_
+#define INTERNET_RADIOBROWSER_RADIOBROWSERSERVICE_H_
+
+#include <QJsonObject>
+
+#include "core/cachedlist.h"
+#include "internet/core/internetmodel.h"
+#include "internet/core/internetservice.h"
+
+class RadioBrowserUrlHandler;
+
+class QNetworkAccessManager;
+class QNetworkReply;
+class QMenu;
+
+class RadioBrowserService : public InternetService {
+  Q_OBJECT
+
+ public:
+  RadioBrowserService(Application* app, InternetModel* parent);
+  ~RadioBrowserService();
+
+  enum ItemType {
+    Type_Stream = 2000,
+  };
+
+  enum Type {
+    Type_Category = InternetModel::TypeCount,
+    Type_CategoryItem,
+    Type_Top100,
+  };
+
+  enum Role {
+    Role_ListUrl = InternetModel::RoleCount,
+    Role_ItemsUrl,
+  };
+
+  struct Stream {
+    QString name_;
+    QUrl url_;
+    QUrl favicon_;
+
+    Song ToSong(const QString& prefix) const;
+  };
+
+  struct Branch {
+    QString name;
+    QString listUrl;
+    QString itemsUrl;
+    Type type;
+  };
+
+  static QList<Branch> BranchList;
+  static QString SearchUrl;
+  static QString PlayClickUrl;
+
+  typedef QList<Stream> StreamList;
+
+  static const char* kServiceName;
+  static const char* kSettingsGroup;
+  static const char* kSchemeName;
+  static const QString defaultServer;
+
+  const QString& url_scheme() const { return url_scheme_; }
+  const QIcon& icon() const { return icon_; }
+  QNetworkAccessManager* network() const { return network_; }
+
+  QStandardItem* CreateRootItem() override;
+  void ShowContextMenu(const QPoint& global_pos) override;
+
+  PlaylistItem::Options playlistitem_options() const override;
+
+  void Search(int search_id, const QString& query, const int limit);
+  void ResolveStationUrl(const QUrl& original_url);
+
+ signals:
+  void SearchFinished(int search_id, RadioBrowserService::StreamList streams);
+  void StationUrlResolved(const QUrl& original_url, const QUrl& url);
+
+ public slots:
+  void ReloadSettings() override;
+
+ private slots:
+  void LazyPopulate(QStandardItem* item) override;
+
+  void RefreshRootItem();
+  void RefreshCategory(QStandardItem* item);
+  void RefreshCategoryItem(QStandardItem* item);
+  void RefreshTop100(QStandardItem* item);
+
+  void RefreshCategoryFinished(QNetworkReply* reply, int task_id,
+                               QStandardItem* item);
+  void RefreshStreamsFinished(QNetworkReply* reply, int task_id,
+                              QStandardItem* item);
+  void ResolveStationUrlFinished(QNetworkReply* reply, int task_id,
+                                 const QUrl& original_url);
+  void SearchFinishedInternal(QNetworkReply* reply, int task_id, int search_id);
+
+  void Homepage();
+  void ShowConfig() override;
+  void AddToSavedRadio(bool checked);
+
+ private:
+  void ReadStation(QJsonObject& value, StreamList* ret);
```

Comment:
I think a lot of these don't need to be member functions and can exist only in the `.cpp` in an anonymous namespace.

Reply:
Moved most private methods in an anonymous namespace; it looked a bit weird to me at first, but it's nice to learn new tricks